lib/internal/main/compile.js

```diff
@@ -0,0 +1,21 @@
+'use strict';
+const {
+  prepareMainThreadExecution,
+  markBootstrapComplete,
+} = require('internal/process/pre_execution');
+const { copyFile, readFile } = require('fs/promises');
+const { emitExperimentalWarning } = require('internal/util');
+const { getOptionValue } = require('internal/options');
+const { inject } = require('internal/deps/postject/postject');
+
+prepareMainThreadExecution(false);
+markBootstrapComplete();
+
+emitExperimentalWarning(`${process.argv0} --compile`);
+
+(async () => {
+  const compilePath = getOptionValue('--compile');
+  const resourceData = await readFile(process.argv[1]);
```

Comment:
Shouldn't this use `resolveMainPath` to be consistant with when the flag is not present?

Reply:
I've removed the entire injector logic for now, so I'm resolving this.